tests/migrations/test_base.py

```diff
@@ -130,3 +132,154 @@ def temporary_migration_module(self, app_label='migrations', module=None):
+            "unique_together": [["pink", "weight"]] if unique_together else [],
+        }
+        if options:
+            model_options["permissions"] = [("can_groom", "Can groom")]
+        if db_table:
+            model_options["db_table"] = db_table
+        operations = [migrations.CreateModel(
+            "Pony",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("pink", models.IntegerField(default=3)),
+                ("weight", models.FloatField()),
+            ],
+            options=model_options,
+        )]
+        if index:
+            operations.append(migrations.AddIndex(
+                "Pony",
+                models.Index(fields=["pink"], name="pony_pink_idx")
+            ))
+        if multicol_index:
+            operations.append(migrations.AddIndex(
+                "Pony",
+                models.Index(fields=["pink", "weight"], name="pony_test_idx")
+            ))
+        if constraints:
+            for constraint in constraints:
+                operations.append(migrations.AddConstraint(
+                    "Pony",
+                    constraint,
+                ))
+        if second_model:
+            operations.append(migrations.CreateModel(
+                "Stable",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                ]
+            ))
+        if third_model:
+            operations.append(migrations.CreateModel(
+                "Van",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                ]
+            ))
+        if related_model:
+            operations.append(migrations.CreateModel(
+                "Rider",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("pony", models.ForeignKey("Pony", models.CASCADE)),
+                    ("friend", models.ForeignKey("self", models.CASCADE))
+                ],
+            ))
+        if mti_model:
+            operations.append(migrations.CreateModel(
+                "ShetlandPony",
+                fields=[
+                    ('pony_ptr', models.OneToOneField(
+                        'Pony',
+                        models.CASCADE,
+                        auto_created=True,
+                        parent_link=True,
+                        primary_key=True,
+                        to_field='id',
+                        serialize=False,
+                    )),
+                    ("cuteness", models.IntegerField(default=1)),
+                ],
+                bases=['%s.Pony' % app_label],
+            ))
+        if proxy_model:
+            operations.append(migrations.CreateModel(
+                "ProxyPony",
+                fields=[],
+                options={"proxy": True},
+                bases=['%s.Pony' % app_label],
+            ))
+        if manager_model:
+            from .models import FoodManager, FoodQuerySet
```

Comment:
@atombrella  Do we need inner imports here? Imports at the top works fine for me.

Reply:
I changed this and pushed some minor edits in the #11687.

Reply:
OK. I think I'll delete that commit from this PR.

Reply:
Done.